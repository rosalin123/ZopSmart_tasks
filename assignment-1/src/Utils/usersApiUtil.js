const baseUrl = 'https://jsonplaceholder.typicode.com';

export const fetchUsers = async () => {
  try {
    let url = `${baseUrl}/users`;
    let users = await fetch(url, { method: 'GET' });
    let usersJson = await users.json();
    return usersJson;
  } catch (err) {
    console.log('Error fetching users details', err.stack);
  }
};

export const getUser = async (id) => {

    let url = `${baseUrl}/users/${id}`;
    let user = await fetch(url, { method: 'GET' });
    let userJson = await user.json();
    return userJson;

};
