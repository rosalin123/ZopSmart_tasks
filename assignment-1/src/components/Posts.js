import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { fetchPosts } from '../actions/postActions';
import { fetchUsers } from '../actions/userActions';
import { connect } from 'react-redux';
import PostsContainer from './postsContainer';
import { clearComments, clearPost } from '../actions/postActions';
import Loader from './loader';
import ErrorComponent from './error';

class Posts extends Component {
  componentDidMount = async () => {
    this.props.fetchUsers();
    this.props.fetchPosts();
    this.props.clearPost();
    this.props.clearComments();
  };
  render() {
    const { loading, posts, error, users } = this.props;

    return loading === true ? (
      <Loader />
    ) : error !== '' ? (
      <ErrorComponent message="Something's not right" />
    ) : (
      <PostsContainer posts={posts} users={users} />
    );
  }
}

const mapStateToProps = (state) => ({
  posts: state.Posts.posts,
  loading: state.Posts.loading,
  error: state.Posts.error,
  users: state.Users.users,
});

const mapDispatchToProps = (dispatch) => {
  return {
    fetchPosts: () => dispatch(fetchPosts()),
    fetchUsers: () => dispatch(fetchUsers()),
    clearComments: () => dispatch(clearComments()),
    clearPost: () => dispatch(clearPost()),
  };
};

Posts.propTypes = {
  posts: PropTypes.array,
  users: PropTypes.array,
};

export default connect(mapStateToProps, mapDispatchToProps)(Posts);
