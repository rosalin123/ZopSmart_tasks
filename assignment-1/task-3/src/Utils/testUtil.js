import React from 'react';

export const findbyTestAttr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};
