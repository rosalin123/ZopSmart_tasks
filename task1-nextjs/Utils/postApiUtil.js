import axios from 'axios';
import fetch from 'isomorphic-unfetch';
const baseUrl = 'https://jsonplaceholder.typicode.com';

export const fetchPosts = async () => {
  let url = `${baseUrl}/posts`;
  let posts = await axios.get(url);
  let postsJson = await posts.json();
  return postsJson;
};

export const getPost = async (id) => {
  let url = `${baseUrl}/posts/${id}`;
  let post = fetch(url);
  let postJson = await post.json();
  return postJson;
};

export const getPostComments = async (id) => {
  let url = `${baseUrl}/posts/${id}/comments`;
  let comments = await axios.get(url);
  let commentsJson = await comments.json();
  return commentsJson;
};

export const getUserPosts = async (id) => {
  let url = `${baseUrl}/posts?userId=${id}`;
  let userPosts = await fetch(url);
  let userPostsJson = await userPosts.json();
  return userPostsJson;
};
