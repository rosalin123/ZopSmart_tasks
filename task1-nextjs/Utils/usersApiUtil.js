import fetch from 'isomorphic-unfetch';
const baseUrl = 'https://jsonplaceholder.typicode.com';

export const fetchUsers = async () => {
  try {
    let url = `${baseUrl}/users`;
    let users = await fetch(url);
    let usersJson = await users.json();
    return usersJson;
  } catch (err) {
    console.log('Error fetching users details', err.stack);
  }
};

export const getUser = async (id) => {
  let url = `${baseUrl}/users/${id}`;
  let user = await fetch(url);
  let userJson = await user.json();
  return userJson;
};
