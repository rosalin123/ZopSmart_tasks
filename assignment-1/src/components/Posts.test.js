import React from 'react';
import { shallow, render } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Posts from './Posts';
import { findfindbyTestAttr, checkProps } from '../Utils/testUtil';
import { indigo } from '@material-ui/core/colors';
import { middleware } from '../store';

const mockStore = configureMockStore(middleware);

const setUp = (initialState) => {
  const store = mockStore(initialState);
  const component = render(
    <Provider store={store}>
      <Posts />
    </Provider>
  );

  return component;
};

describe('Posts Component', () => {
  let component;

  describe('Checking prop types', () => {
    it('should not return a warning', () => {
      const expectedProps = {
        posts: ['post1', 'post2'],
        users: ['user1', 'user2'],
      };

      const propsErr = checkProps(Posts, expectedProps);
      expect(propsErr).toBeUndefined();
    });
  });

  it('Posts should render without throwing an error', () => {
    const initialState = {
      Posts: { loading: false, error: '', posts: [] },
      Users: { loading: false, error: '', users: [] },
    };
    component = setUp(initialState);
    expect(component).toMatchSnapshot();
  });
});
