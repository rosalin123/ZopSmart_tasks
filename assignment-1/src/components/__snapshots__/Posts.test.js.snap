// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Component Posts should render without throwing an error 1`] = `
<div
  class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  data-test="postContainerComponent"
>
  <div
    class="MuiGrid-root MuiGrid-item"
    data-test="header"
  >
    <div
      class="makeStyles-headerStyles-2"
    >
      Posts
    </div>
  </div>
</div>
`;
